packages/arb-bridge-eth/contracts/rollup/Rollup.sol

```diff
@@ -0,0 +1,699 @@
+                assertion.gasUsed >= timeSinceLastNode * arbGasSpeedLimitPerBlock,
+            "TOO_SMALL"
+        );
+
+        // Don't allow an assertion to use above a maximum amount of gas
+        require(assertion.gasUsed <= timeSinceLastNode * arbGasSpeedLimitPerBlock * 4, "TOO_LARGE");
+
+        uint256 deadlineBlock = block.number + challengePeriodBlocks;
+        uint256 prevDeadlineBlock = prevNode.deadlineBlock();
+        if (deadlineBlock < prevDeadlineBlock) {
+            deadlineBlock = prevDeadlineBlock;
+        }
+        deadlineBlock += assertion.gasUsed / arbGasSpeedLimitPerBlock;
+
+        INode node =
+            INode(
+                nodeFactory.createNode(
+                    RollupLib.nodeStateHash(assertion, inboxMaxCount),
+                    RollupLib.challengeRoot(
+                        assertion,
+                        inboxMaxCount,
+                        inboxMaxAcc,
+                        assertion.gasUsed / arbGasSpeedLimitPerBlock
+                    ),
+                    RollupLib.confirmHash(assertion),
+                    latestStakedNode(msg.sender),
+                    deadlineBlock
+                )
+            );
+
+        nodeCreated(node);
+        stakeOnNode(msg.sender, nodeNum);
+
+        emit NodeCreated(
+            nodeNum,
+            assertionBytes32Fields,
+            assertionIntFields,
+            inboxMaxCount,
+            inboxMaxAcc
+        );
+    }
+
+    /**
+     * @notice Refund a staker that is currently staked on or before the latest confirmed node
+     * @param stakerAddress Address of the staker whose stake is refunded
+     */
+    function returnOldDeposit(address stakerAddress) external override whenNotPaused {
+        require(latestStakedNode(stakerAddress) <= latestConfirmed(), "TOO_RECENT");
+        checkUnchallengedStaker(stakerAddress);
+        withdrawStaker(stakerAddress);
+    }
+
+    /**
+     * @notice Increase the amount staked for the given staker
+     * @param stakerAddress Address of the staker whose stake is increased
+     * @param tokenAmount If staking in something other than eth, this is the amount of tokens staked, otherwise 0
+     */
+    function addToDeposit(address stakerAddress, uint256 tokenAmount)
+        external
+        payable
+        whenNotPaused
+    {
+        checkUnchallengedStaker(stakerAddress);
+        increaseStakeBy(stakerAddress, receiveStakerFunds(tokenAmount));
+    }
+
+    /**
+     * @notice Reduce the amount staked for the sender
+     * @param target Target amount of stake for the staker. If this is below the current minimum, it will be set to minimum instead
+     */
+    function reduceDeposit(uint256 target) external whenNotPaused {
+        checkUnchallengedStaker(msg.sender);
+        uint256 currentRequired = currentRequiredStake();
+        if (target > currentRequired) {
```

Comment:
I think this should be `target < currentRequired`.

Reply:
Fixed